lib/libutils/ext/ftrace/ftrace.c

```diff
@@ -253,54 +255,57 @@ unsigned long __noprof ftrace_return(void)
 	else
 		return 0;
 
-	curr_buf = (char *)fbuf + fbuf->buf_off + fbuf->curr_size;
+	/* Check if we have a minimum ftrace buffer current size */
+	line_size = ENTRY_SIZE(fbuf->ret_idx);
+	if (fbuf->curr_size < line_size)
+		goto out;
```

Comment:
I don't understand why returning `fbuf->ret_stack[fbuf->ret_idx]` here.

Reply:
This is to ensure even if we have somehow corrupted ftrace buffer formatting, the function call chain shouldn't get broken since we have a valid `fbuf->ret_idx` at this point.

Reply:
I think it would be better to explicit that in an inline comment. I would help people understanding what's happening here.

Reply:
Sure.

Reply:
thanks